### PR TITLE
[FA] Upstream FA qk initialization

### PR DESCRIPTION
**Pull request by binarman**
This PR replaces qk matrix initialization with
upstream version

### CODE REVIEW

**Comment by binarman** [comment]
@vgokhale
During IFU, I believed that old version of qk initialization is something that we introduced in ROCm fork, because it produced better code.

Now I see that it is actually an old version from upstream, which was changed, so this PR adjusts our code with upstream.

File: python/tutorials/06-fused-attention.py [hunk 1]
```diff
@@ -200,10 +200,10 @@ def _bwd_kernel(
             q = tl.load(q_ptrs)
             # recompute p = softmax(qk, dim=-1).T
             if CAUSAL:
-                qk = tl.dot(q, tl.trans(k), out_dtype=tl.float32)
-                qk = tl.where(offs_m_curr[:, None] >= (offs_n[None, :]), qk, float("-inf"))
+                qk = tl.where(P_SEQ + offs_m_curr[:, None] >= (offs_n[None, :]), float(0.), float("-inf"))
             else:
-                qk = tl.dot(q, tl.trans(k), out_dtype=tl.float32)
+                qk = tl.zeros([BLOCK_M, BLOCK_N], dtype=tl.float32)
+            qk += tl.dot(q, tl.trans(k))
```
Comment (vgokhale) [inline python/tutorials/06-fused-attention.py:206]: Nitpicky comment, but do you think it'd be cleaner to do it how the fwd kernel does it at line 91 above?

**Review by vgokhale — APPROVED** [review]

File: python/tutorials/06-fused-attention.py (hunk 1 above)
Comment (binarman) [reply python/tutorials/06-fused-attention.py:206]: Actually, I like it in fwd better. But intention of this PR is to align our fork with upstream, to make IFUs and stuff easier, so I just copied what we have in upstream.

Comment (binarman) [reply python/tutorials/06-fused-attention.py:206]: Not sure, if there any difference in term of performance, though.
We should check it, I think.

I believe we do not use this version for benchmarking, so I did not check it yet.

**Pull request merged**